fixed-hash/src/hash.rs

```diff
@@ -322,6 +367,10 @@ macro_rules! construct_hash {
 			}
 		}
 
+		#[deprecated(
+			since = "0.2.5",
+			note = "no proper error handling possible with out-of-bounds inputs"
```

Comment:
Remind me, what is the replacement for this in `0.3`? I could be wrong but I think this one is fairly often used.

Reply:
I had no concrete replacement API in mind for this but I would be O.K. with replacing this by a function:

```rust
fn from_slice(slice: &[u8]) -> $from {
    assert_eq!(
        $from::len_bytes(), slice.len(),
        "[fixed-hash] error: cannot create a fixed-hash from a byte slice with different length"
    );
    $from::from(std::mem::transmute::<&[u8], &[u8; $n_bytes]>(slice))
}
```

Of course I really hope for not having to use `mem::transmute`.